0000-ilp-ping-protocol/0000-ilp-ping-protocol.md

```diff
@@ -0,0 +1,78 @@
+
+Instead, this protocol uses only [Interledger-layer](https://github.com/interledger/rfcs/blob/master/0001-interledger-architecture/0001-interledger-architecture.md#interledger-protocol) technologies, plus a known condition/fulfillment pair, to test connectivity between two nodes in the Interledger.
+ 
+The ping protocol is not designed to debug routing issues and does not provide additional diagnostic information about the state of routing. That use case would be better served by a separate `traceroute` mechanism.
+
+All Interledger implementations _SHOULD_ respond to ping protocol requests, unless a node has a specific reason not to, such as security or privacy concerns.
+
+## Overview
+This RFC defines two ping protocol modes: unidirectional and bidirectional.
+
+* **Unidirectional**: A ping Initiator sends a specially formed ILP Prepare packet, and the Receiver either fulfills or rejects the packet according to the rules defined in [Unidirectional Mode](#Unidirectional Mode).
+* **Bidirectional**: An Initiator sends a specially formed "ping" packet that prompts the Receiver to send an additional "pong" packet. Upon receiving this pong packet, the Initiator either fulfills or rejects according to the rules defined in [Bidirectional Mode](#Bidirectional Mode), thus prompting the Receiver to fulfill or reject the original ping packet.
+
+The primary difference between the two modes is that unidirectional mode allows an Initiator to test connectivity from the Initiator to the Recipient only, whereas bidirectional mode allows the Initiator to test connectivity in both directions.
+
+## Terminology
+
+* The **Initiator** is the Interledger node initiating the ping request.
+* The **Recipient** is the Interledger node responding to the Initiator's request.
+* **Source amount** is the amount debited from the sender of an ILP payment.
+* **Destination amount** is the amount credited to the Recipient of an ILP payment.
+* The **Known Preimage** is the [ASCII](https://tools.ietf.org/html/rfc20) string `pingpingpingpingpingpingpingping`, which in binary is `70696E6770696E6770696E6770696E6770696E6770696E6770696E6770696E67`. This value is used in Unidirectional mode only. 
+* The **Known Condition** is the SHA256 hash of the Known Preimage, which when encoded using Base64 is `jAC8DGFPZPfh4AtZpXuvXFe2oRmpDVSvSJg2oT+bx34=`. This value is used in Unidirectional mode only.
+
+## Unidirectional Mode
+
+1. Initiator sends a ILP Prepare packet using the following details:
+   - **Destination**: Recipient's Interledger address.
+   - **Amount**: Any amount chosen by the Initiator.
+   - **Expiry**: Any appropriate expiry.
```

Comment:
Want to give a recommendation here? If I were implementing this, I might not know what an appropriate expiry might be

Reply:
connector.land, moneyd-gui and ilp-ping implement the bidirectional mode and have a default expiration set to 30000ms. So maybe a recommendation could be 15000ms for the unidirectional mode.

Reply:
After considering this change for a while, I don't think the RFC should really take a position on a recommended expiry. 

I say this because I was about to write, "This value will vary by path, but should generally not exceed 30 seconds." But then, after seeing it in there, it felt artificially proscriptive. For example, there might be times where this value should not exceed some other value, like maybe 5 seconds.

Personally, I like the text as-is here - it's basically saying that the initiator can pick whatever value they find appropriate. For very long paths, I supposed there might even be a time where 30s isn't enough.